src/domain/prices/prices.repository.ts

```diff
@@ -0,0 +1,63 @@
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  ILoggingService,
+  LoggingService,
+} from '../../logging/logging.interface';
+import { IPricesApi } from '../interfaces/prices-api.interface';
+import { IPricesRepository } from './prices.repository.interface';
+import { isArray } from 'lodash';
+
+@Injectable()
+export class PricesRepository implements IPricesRepository {
+  constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(IPricesApi) private readonly pricesApi: IPricesApi,
+  ) {}
+
+  async getNativeCoinPrice(args: {
+    nativeCoinId: string;
+    fiatCode: string;
+  }): Promise<number> {
+    try {
+      const result = await this.pricesApi.getNativeCoinPrice(args);
+      const { nativeCoinId, fiatCode } = args;
+      const nativeCoinPrice = result?.[nativeCoinId]?.[fiatCode];
+      if (!nativeCoinPrice) {
+        this.loggingService.error(`Got an invalid ${nativeCoinId} price`);
+        return 0;
+      }
+      return nativeCoinPrice;
+    } catch (err) {
+      this.loggingService.error(err.message);
+      return 0;
+    }
+  }
+
+  async getTokenPrice(args: {
+    chainName: string;
+    tokenAddress: string;
+    fiatCode: string;
+  }): Promise<number> {
+    try {
+      const result = await this.pricesApi.getTokenPrice(args);
+      const { tokenAddress, fiatCode } = args;
+      const tokenPrice = result?.[tokenAddress.toLowerCase()]?.[fiatCode];
+      if (!tokenPrice) {
+        this.loggingService.warn(`Got an invalid ${tokenAddress} price`);
+        return 0;
+      }
+      return tokenPrice;
+    } catch (err) {
+      this.loggingService.error(err.message);
+      return 0;
+    }
+  }
+
+  async getFiatCodes(): Promise<string[]> {
+    const fiatCodes = await this.pricesApi.getFiatCodes();
+    if (!isArray(fiatCodes)) {
```

Comment:
Wouldn't AJV be useful to tackle the validation of these entities?

Reply:
Yes, good point. This should be just a string array, but for consistency we should declare a schema and use it. Thanks!

Reply:
Done [17040f3](https://github.com/safe-global/safe-client-gateway/pull/732/commits/17040f3f07ad63a3682ce6b665dc451dfae545a2)